contracts/contracts/strategies/BaseTwoAssetCurveStrategy.sol

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title Curve Strategy for pools with twp assets
+ * @notice Investment strategy for investing assets via a Curve pool. eg frxETH/ETH
+ * @dev There are a number of restrictions on which Curve pools can be used by this strategy
+ * - all assets have the same decimals as the Curve pool's LP token. eg 18
+ * - the Curve pool and Curve pool LP token are the same contract.
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { ICurveMetaPool } from "./ICurveMetaPool.sol";
+import { IERC20, InitializableAbstractStrategy } from "../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../utils/StableMath.sol";
+
+abstract contract BaseTwoAssetCurveStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 public constant MAX_SLIPPAGE = 1e16; // 1%, same as the Curve UI
+    /// @notice number of assets in base Curve pool
+    uint256 public constant CURVE_BASE_ASSETS = 2;
+
+    int256[49] private __reserved;
+
+    /**
+     * @dev Deposit asset into the Curve pool
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_amount > 0, "Must deposit something");
+        emit Deposit(_asset, platformAddress, _amount);
+
+        // Curve requires passing deposit amounts for all assets
+        uint256[CURVE_BASE_ASSETS] memory _amounts = [uint256(0), 0];
+        // Validate the asset and get Curve pool index position
+        uint256 poolCoinIndex = _getCoinIndex(_asset);
+        // Set the amount on the asset we want to deposit
+        _amounts[poolCoinIndex] = _amount;
+        ICurveMetaPool curvePool = ICurveMetaPool(platformAddress);
+        uint256 depositValue = _amount.divPrecisely(
```

Comment:
BaseCurveStrategy has to scale USDT and USDC up to 18 decimals to match DAI and the Curve pool LP token.
For the ETH-based Curve pools, all the assets are 18 decimals so they don't need scaling up.
I commented on the contract that the strategy can only be used if all the assets and LP token have the same decimals.
This might get added back if I can make `BaseCurveStrategy` handle Curve pools with either 2 or 3 assets.

Reply:
thanks that makes sense